src/core/thread/dua_manager.cpp

```diff
@@ -66,33 +68,110 @@ void DuaManager::UpdateDomainUnicastAddress(BackboneRouter::Leader::DomainPrefix
         Get<ThreadNetif>().RemoveUnicastAddress(mDomainUnicastAddress);
     }
 
-    VerifyOrExit(aState == BackboneRouter::Leader::kDomainPrefixAdded ||
-                     (aState == BackboneRouter::Leader::kDomainPrefixRefreshed),
+    VerifyOrExit(aState == BackboneRouter::Leader::kDomainPrefixRefreshed ||
+                     aState == BackboneRouter::Leader::kDomainPrefixAdded,
                  OT_NOOP);
 
     prefix = Get<BackboneRouter::Leader>().GetDomainPrefix();
 
     OT_ASSERT(prefix != NULL);
 
-    mDomainUnicastAddress.GetAddress().Clear();
-    mDomainUnicastAddress.GetAddress().SetPrefix(prefix->mPrefix.mFields.m8, prefix->mLength);
     mDomainUnicastAddress.mPrefixLength = prefix->mLength;
 
-    if (Get<Utils::Slaac>().GenerateIid(mDomainUnicastAddress, NULL, 0, &mDadCounter,
-                                        false /* not allow random IID if no valid IID */) == OT_ERROR_NONE)
+    // Apply cached DUA manually configured when prefixes match.
+    if (!mFixedDomainUnicastAddress.IsUnspecified() &&
+        Ip6::Address::PrefixMatch(mFixedDomainUnicastAddress.mFields.m8, prefix->mPrefix.mFields.m8, prefix->mLength) >=
+            prefix->mLength)
     {
-        mDomainUnicastAddress.mValid = true;
+        mDomainUnicastAddress.GetAddress() = mFixedDomainUnicastAddress;
+        mDomainUnicastAddress.mValid       = true;
         Get<ThreadNetif>().AddUnicastAddress(mDomainUnicastAddress);
     }
     else
+    {
+        mDomainUnicastAddress.GetAddress().Clear();
```

Comment:
Not directed related to this PR, however I think we should set DUA to be invalid here. 
```suggestion
        mDomainUnicastAddress.GetAddress().Clear();
        mDomainUnicastAddress.mValid = False;
```